### PR TITLE
Modify purge-job logic to store old documents to Ceph

### LINKED ISSUE #1516: Extend purge-job logic to handle documents placed on Ceph
**Is your feature request related to a problem? Please describe.**

As Thoth administrator, I would like to make sure purge-job logic takes care of old documents stored on Ceph. This way we will keep only relevant JSON data present on Ceph under corresponding key (e.g. sync-job will not pick them up and sync them again once triggered).

**Describe the solution you'd like**

- [ ] use thoth-storages to query for solver document ids that are about to be deleted during purge
- [ ] move solver documents to another location (e.g. store them under another prefix)
   - [ ] handle adviser documents
   - [ ] handle solver documents
   - [ ] handle container image analyses documents
- [ ] remove entries from the database once all the relevant documents have been moved to another location
- [x] keep "to be depracated" solvers to the configmap
- [x] adviser notifies about soon to be depricated solvers
- [ ] update solver procedure document and state steps that need to be done each time solvers are removed/added - https://github.com/thoth-station/thoth-application/blob/master/docs/thoth_s2i_procedure.md

**Describe alternatives you've considered**

We could also delete the documents, but they can be a source of interesting data later (or have them available for experiments).

### PR DESCRIPTION
**Pull request by mayaCostantini**
## Related Issues and Dependencies

Fixes: https://github.com/thoth-station/thoth-application/issues/1516

## This introduces a breaking change

- [x] No

## This Pull Request implements

Applies the store_old_documents() method to solver, adviser and package-extract documents.

## Description

store_old_documents() has been implemented to move solver documents that are about to be deleted by purge-job to Ceph using thoth-storages tools.

### CODE REVIEW

**Review by fridex — COMMENTED** [review]
The approach looks good to me 👍🏻  Let's wait for the subsequent changes based on https://github.com/thoth-station/storages/pull/2468. We will also need a new thoth-storages release to have this support. So, let's wait with merge for now

/hold

File: app.py [hunk 1]
```diff
@@ -104,6 +105,7 @@ def solver(os_name: str, os_version: str, python_version: str) -> None:
     """Purge solver data."""
     graph = GraphDatabase()
     graph.connect()
+    graph.purge_solver_documents(os_name=os_name, os_version=os_version, python_version=python_version)
     result = graph.delete_solved(os_name=os_name, os_version=os_version, python_version=python_version)
```
Comment (fridex) [inline app.py]: If I read this correctly, this line can be dropped completely as the logic behind `purge_solver_documents` handles deletion:

```suggestion
```

Comment (mayaCostantini) [reply app.py]: Correct, I will remove this line

**Review by harshad16 — COMMENTED** [review]
/deploy

**Review by fridex — COMMENTED** [review]
Changes look good to me 👍🏻 Let's wait for https://github.com/thoth-station/storages/issues/2497 and we can have this in 💯 Thanks!

**Review by fridex — APPROVED** [review]
looks really good, thanks 👍🏻

**Comment by mayaCostantini** [comment]
> looks really good, thanks 👍🏻

Thank you!

**Comment by fridex** [comment]
@mayaCostantini can we unhold this?

**Comment by mayaCostantini** [comment]
> @mayaCostantini can we unhold this?

Sure
/unhold

**Pull request merged**